### DISCUSSION TITLE
Pagination shows too many pages relative to siblingCount

### DISCUSSION THREAD

**Discussion by Jonas-C** [chakra-ui/zag#1890] [Q&A] [answered]
# 🐛 Bug report

The value of `siblingCount` doesn't seem to represent the actual number of pagination items that are shown. The official example defaults to `1`, but six items are shown. Shouldn't this actually show three items? If I set `siblingCount` to 0 I'd expect only the current page to render. I'm assuming this is related to the position of the current page?

## 💥 Steps to reproduce

1. Go to https://zagjs.com/components/react/pagination
2. See error

## 🧐 Expected behavior

I'd expect the `siblingCount` to accurately describe the amount of visible siblings, regardless of the active page. 

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      |  0.71.0          |
| Browser          | Firefox           |
| Operating System | MacOS           |

**Comment by segunadebayo** [comment, marked as answer]
This is by design. Sibling count is taken into account when we get to the “ellipsis” mode and needs to render at least one item to the left or right.

For UX and SEO reasons, it’s recommended to render the first, last, previous button, and next. Adding to that the sibling count, gets to around 5-6 items shown per time.

This prop doesn’t strictly mean “always show X items”, what you’re looking might need to be handled in a different way.

Sibling and Boundaries are two properties that a pagination could have to absolutely control the number of visible items. I don’t think we implement boundaries at the monent.

**Reply by Kavian77** [reply]
Thanks, this clarifies. To follow up:  
To work around this, is it perhaps possible to specify a maximum number of items rendered (either including or excluding the ellipsis items) to avoid getting more items rendered than what is needed by the consumer?
